### PR TITLE
Fix bulk send

### LINKED ISSUE #1283: Bug: BulkSend doesn't work.
**Describe the bug**
Outside the USA, when a volunteer clicks the `Send Bulk` button on the texter view, it correctly calls the `bulkSendMessages` mutation but the implementation of the mutation raises an exception and sends nothing.

**To Reproduce**
1. Set up Spoke for bulk sending.
2. Ensure that the user has an assignment with more than one contact.
3. In the texter contact screen, click `Send Bulk`.

**Expected behavior**
Multiple messages are sent.

**Screenshots**
N/A

**Additional context**
This is what we see in the log:

```
2019-11-21T01:52:44.577545+00:00 app[web.1]: (node:14) UnhandledPromiseRejectionWarning: TypeError: _campaignContact.resolvers.CampaignContact.currentInteractionStepScript is not a function
2019-11-21T01:52:44.577565+00:00 app[web.1]:     at _callee33$ (/spoke/build/server/server/api/schema.js:2077:79)
2019-11-21T01:52:44.577567+00:00 app[web.1]:     at tryCatch (/spoke/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
2019-11-21T01:52:44.577569+00:00 app[web.1]:     at Generator.invoke [as _invoke] (/spoke/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:296:22)
2019-11-21T01:52:44.577570+00:00 app[web.1]:     at Generator.prototype.(anonymous function) [as next] (/spoke/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:114:21)
2019-11-21T01:52:44.577572+00:00 app[web.1]:     at step (/spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
2019-11-21T01:52:44.577574+00:00 app[web.1]:     at /spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
2019-11-21T01:52:44.577575+00:00 app[web.1]:     at new Promise (<anonymous>)
2019-11-21T01:52:44.577576+00:00 app[web.1]:     at new F (/spoke/node_modules/core-js/library/modules/_export.js:36:28)
2019-11-21T01:52:44.577578+00:00 app[web.1]:     at /spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
2019-11-21T01:52:44.577579+00:00 app[web.1]:     at /spoke/build/server/server/api/schema.js:2107:35
2019-11-21T01:52:44.577581+00:00 app[web.1]:     at Array.map (<anonymous>)
2019-11-21T01:52:44.577582+00:00 app[web.1]:     at _callee34$ (/spoke/build/server/server/api/schema.js:2069:33)
2019-11-21T01:52:44.577583+00:00 app[web.1]:     at tryCatch (/spoke/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:62:40)
2019-11-21T01:52:44.577585+00:00 app[web.1]:     at Generator.invoke [as _invoke] (/spoke/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:296:22)
2019-11-21T01:52:44.577586+00:00 app[web.1]:     at Generator.prototype.(anonymous function) [as next] (/spoke/node_modules/babel-runtime/node_modules/regenerator-runtime/runtime.js:114:21)
2019-11-21T01:52:44.577587+00:00 app[web.1]:     at step (/spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
2019-11-21T01:52:44.577589+00:00 app[web.1]:     at /spoke/node_modules/babel-runtime/helpers/asyncToGenerator.js:28:13
2019-11-21T01:52:44.577733+00:00 app[web.1]: (node:14) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
```

### PR DESCRIPTION
**Pull request by lperson**
# Fixes #1283 

## Description

This PR fixes bulk send, which is used outside the USA to send more than one initial message with a single button click.  It hadn't gotten any love in a while and when Australian Youth Climate Action tried to use the feature, no messages got sent. 

This is bigger than 300 lines because I moved 3 large functions out of rootMutations in `src/server/api/schema.js` to improve readability.

#### New folder
`src/server/api/mutations'

##### New files

- `src/server/api/mutations/bulkSendMessages.js` -- Move the functionality for the bulkSendMessage mutation from `src/server/api/schema.js` to this file.  **_The fix for issue #1282 was done here.  Other than tests, this is the only place where there was a change in functionality._**
- `src/server/api/mutations/findNewCampaignContact.js` -- Move the functionality for the findNewCampaignContact mutation from `src/server/api/schema.js` to this file.  This functionality did not change.
- `src/server/api/mutations/sendMessage.js` -- Move the functionality for the sendMessage mutation from `src/server/api/schema.js` to this file.  This functionality did not change.
- `index.js` -- imports and exports

### How the bug was fixed

Before, the `bulkSendMessages` mutation was calling a non-existent method (`currentInteractionStepScript`) on each contact to get the initial message for each contact.  The fix is to get the first step in the interaction hierarchy and use the `applyScript` method to personalize it for each contact.  The change was done in `src/server/api/mutations/bulkSendMessages.js`.

# Checklist:

- [N/A]  ~I have manually tested my changes on desktop and mobile~
- [x] The test suite passes locally with my changes
- [N/A] ~If my change is a UI change, I have attached a screenshot to the description section of this pull request~
- [x] [My change is 300 lines of code or less](https://github.com/MoveOnOrg/Spoke/blob/main/CONTRIBUTING.md#your-first-code-contribution), or has a documented reason in the description why it’s longer
- [N/A ] ~I have made any necessary changes to the documentation~
- [x] I have added tests that prove my fix is effective or that my feature works
- [N/A ] ~My PR is labeled [WIP] if it is in progress~

### CODE REVIEW

File: __test__/server/api/campaign.test.js [hunk 1]
```diff
@@ -778,6 +779,82 @@ describe("Reassignments", async () => {
...
+    );
+
+    // TEXTER 1 (100 needsMessage)
+    expect(texterCampaignDataResults.data.assignment.contacts.length).toEqual(
+      100
+    );
+    expect(texterCampaignDataResults.data.assignment.allContactsCount).toEqual(
+      100
+    );
+    console.timeEnd("func1");
```
Comment (schuyler1d) [inline __test__/server/api/campaign.test.js]: remove `console.time*` functions here please (I accidentally left this in -- but am removing them from my previous tests soon)

File: __test__/server/api/campaign.test.js [hunk 2]
```diff
@@ -778,6 +779,82 @@ describe("Reassignments", async () => {
     expect(texterCampaignDataResults2.data.assignment.allContactsCount).toEqual(
       29
     );
-    console.timeEnd('fullfunction');
-  }, 10000);  // long test can exceed default 5seconds
+    console.timeEnd("fullfunction");
+  }, 10000); // long test can exceed default 5seconds
+});
+
+describe("Bulk Send", async () => {
```
Comment (schuyler1d) [inline __test__/server/api/campaign.test.js:792]: can you add a test above here that confirms that when NOT_IN_USA and/or ALLOW_SEND_ALL are not enabled that bulkSend fails or errors?

**Review by schuyler1d — CHANGES_REQUESTED** [review]
I like the move from the overly enormous schema.js of these important and complex mutations.
This does make me a little nervous about the difficultly of merging this with other changes in a batch, but that can be the challenge of our staging merge.

I asked for a couple of test tweaks.  Can you also include a description of what you actually changed to fix this?  That's the other issue with the single diff that moves everything -- it's impossible to see what changed, so let's at least describe in the PR.  (in the future, please make a separate commit for changes vs. code movement/refactoring)

**Comment by lperson** [comment]
@schuyler1d Thanks for the review.  This is ready to re-review.

**Comment by schuyler1d** [comment]
Cool.  These are good tests and I can approve the content of the PR.
@lperson Can you get the tests passing?

**Comment by schuyler1d** [comment]
@ibrand Please put/keep this PR 'on deck' for stage-main merge when tests are passing -- as a pre-step, would be useful to get reports on what files specifically conflict.  I believe my PR will but just to add a comment into the sendMessage call:
https://github.com/MoveOnOrg/Spoke/pull/1273/files#diff-e3e0c0fe13e813c7748359d6a931a6faR1163

File: __test__/server/api/campaign.test.js [hunk 3]
```diff
@@ -778,6 +781,140 @@ describe("Reassignments", async () => {
...
+    expect(texterCampaignDataResults.data.assignment.contacts.length).toEqual(
+      NUMBER_OF_CONTACTS - expectedSentCount
+    );
+    expect(texterCampaignDataResults.data.assignment.allContactsCount).toEqual(
+      NUMBER_OF_CONTACTS
+    );
+  };
+
+  const expectErrorBulkSending = (result) => {
+    expect(result.errors[0]).toBeDefined();
```
Comment (schuyler1d) [inline __test__/server/api/campaign.test.js:867]: It's fine to make this ambiguous if it's not consistent.  Can you add a comment on what it outputs?

**Review by schuyler1d — DISMISSED** [review]
:+1:

**Review by schuyler1d — APPROVED** [review]
:+1:

**Pull request merged**